src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -130,11 +130,28 @@ private MessageCacheItem createCacheItem(Iterator<Locale> fallbackLocalesIter) {
 		CacheService cacheService = new CacheService(dto);
 		// Create a new cacheItem object to be stored in cache
 		MessageCacheItem cacheItem = new MessageCacheItem();
-		refreshCacheItem(cacheItem, VIPCfg.getInstance().getMsgOriginsQueue().iterator());
 
+		// If the requested locale is not supported, but matches a supported locale (eg. requested locale "fr_CA matches supported locale "fr"),
+		// return the messages of the supported locale that best matches the requested locale.
+		ProductService ps = new ProductService(dto);
+		if (!ps.isSupportedLocale(Locale.forLanguageTag(dto.getLocale()))) {
+			Locale matchedLocale = LocaleUtility.pickupLocaleFromList(new LinkedList<>(ps.getSupportedLocales()), Locale.forLanguageTag(dto.getLocale()));
+			if (!matchedLocale.equals(Locale.forLanguageTag(dto.getLocale()))) {
```

Comment:
I think this judgement may not needed, because no matter there is matched locale or not, below code need be executed. 
for example, fr supported, no matter request fr-CA or ru, below code should be executed.